File: source/aoBlogs2/Models/GroupRuleModel.vb

Option Explicit On
Option Strict On

Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Contensive.BaseClasses

Namespace Models
    Public Class GroupRuleModel
        Inherits DbModel
        '
        '====================================================================================================
        '-- const
        Public Const contentName As String = "Group Rules"      '<------ set content name
        Public Const contentTableName As String = "xxxxxtableNameGoesHerexxxxx"   '<------ set to tablename for the primary content (used for cache names)
        Private Shadows Const contentDataSource As String = "ccGroupRules"             '<------ set to datasource if not default
        '
        '====================================================================================================
        ' -- instance properties

        Public Property AllowAdd As Boolean
        Public Property AllowDelete As Boolean
        Public Property ContentID As Integer
        Public Property GroupID As Integer

    End Class
End Namespace

File: source/aoBlogs2/Models/Db/GroupRuleModel.cs


//using Contensive.Models.Db;

//namespace Contensive.Addons.Blog.Models {
//    public class GroupRuleModel : Contensive.Models.Db.DbBaseModel {
//        // 
//        // ====================================================================================================
//        public static DbBaseTableMetadataModel tableMetadata { get; private set; } = new DbBaseTableMetadataModel("Group Rules", "xxxxx", "default", false);
//        // -- const
//        public const string contentName = "Group Rules";      // <------ set content name
//        public const string contentTableName = "xxxxxtableNameGoesHerexxxxx";   // <------ set to tablename for the primary content (used for cache names)
//        private  const string contentDataSource = "ccGroupRules";             // <------ set to datasource if not default
//        // 
//        // ====================================================================================================
//        // -- instance properties

//        public bool AllowAdd { get; set; }
//        public bool AllowDelete { get; set; }
//        public int ContentID { get; set; }
//        public int GroupID { get; set; }

//    }
//}